sycl/test/aot/spirv-fallback.cpp

```diff
@@ -0,0 +1,107 @@
+// REQUIRES: ioc64, ocloc, aoc
```

Comment:
Thanks for the tests! :)
But I see there a lot of code duplication, I think we can merge multiple-devices and spirv-fallback tests at least.

Reply:
The separation of certain tests brings in the code duplication, but also allows to isolate issues with a particular AOT tool/device. E.g. if no ioc64 is installed on the machine, GPU and FPGA still have some coverage.

That said, I agree that merging the "all tools" tests is needed.

Reply:
Done